internals/cli/cmd_run.go

```diff
@@ -202,13 +211,21 @@ func runDaemon(rcmd *cmdRun, ch chan os.Signal, ready chan<- func()) error {
 		} else {
 			logger.Noticef("Started default services with change %s.", changeID)
 		}
-	}
 
-	var stop chan struct{}
-	if ready != nil {
-		stop = make(chan struct{}, 1)
-		ready <- func() { close(stop) }
-		close(ready)
+		if ready != nil {
+			// wait for the default services to start
+			autoStartReady = make(chan struct{}, 1)
+			go func() {
+				waitCmd := waitMixin{
+					hideProgress: true,
+				}
+				waitCmd.setClient(rcmd.client)
+				waitCmd.wait(changeID)
```

Comment:
The wait call can fail. The signature of `wait()` is
```
func (wmx waitMixin) wait(id string) (*client.Change, error) {
```
While the commit message makes it clear that we ignore `client.Change.Err != ""`, I think we should fail on non-nil error, because the cause may lead to further errors that will be hard to debug when we silently ignore the original error.

Reply:
Thinking more about it: If we fail here, we have to stop the daemon gracefully (i.e. with services) as well. If that turns out to be too complicated, we should at least log the error from `wait()`. But there has to be more places where we fail on an error and I suspect the daemon is going to be stopped is in some `defer`-ed function.

Reply:
Hiya, I took the easier route and just logged the error message for now. Let me know if that's okay.